### PR TITLE
Single multiple question api

### PR DESCRIPTION
**Pull request by viharsanghavi**
Redmine Task: #20902
File Modified:
QuestionController.cs
IQuestionRepository.cs
QuestionRepository.cs

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
 
 namespace Promact.Trappist.Core.Controllers
 {
-    [Route("api/Questions")]
+    [Route("api")]
     public class QuestionsController : Controller
     {
         private readonly IQuestionsRespository _questionsRepository;
-
         public QuestionsController(IQuestionsRespository questionsRepository)
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs]: the name should be singular

File: Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs [hunk 2]
```diff
@@ -1,28 +1,30 @@
...
+        [Route("single-multiple-question")]
+        [HttpPost]
+        public IActionResult AddSingleMultipleAnswerQuestion(SingleMultipleAnswerQuestion singleMultipleAnswerQuestion, SingleMultipleAnswerQuestionOption singleMultipleAnswerQuestionOption)
         {
-            var questions = _questionsRepository.GetAllQuestions();
-
-            return Json(questions);
-        }
+            _questionsRepository.AddSingleMultipleAnswerQuestion(singleMultipleAnswerQuestion, singleMultipleAnswerQuestionOption);
+            return Ok();
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs]: return type is not valid

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionsRepository.cs [hunk 3]
```diff
@@ -20,9 +19,19 @@ public QuestionsRepository(TrappistDbContext dbContext)
...
         }
+        /// <summary>
+        /// Add single multiple answer question into SingleMultipleAnswerQuestion model
+        /// </summary>
+        /// <param name="singleMultipleAnswerQuestion"></param>
+        /// <param name="singleMultipleAnswerQuestionOption"></param>
+        public void AddSingleMultipleAnswerQuestion(SingleMultipleAnswerQuestion singleMultipleAnswerQuestion, SingleMultipleAnswerQuestionOption singleMultipleAnswerQuestionOption)
+        {
+            _dbContext.SingleMultipleAnswerQuestion.Add(singleMultipleAnswerQuestion);
+            _dbContext.SingleMultipleAnswerQuestionOption.Add(singleMultipleAnswerQuestionOption);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionsRepository.cs]: who will enter data in mapping table?

Comment (viharsanghavi) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionsRepository.cs]: Sorry I can't understand your question

File: Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
         /// </summary>
-        /// <returns>Questions list</returns>
-        [HttpGet]
-        public IActionResult GetQuestions()
+        /// <param name="singleMultipleAnswerQuestion"></param>
+        /// <param name="singleMultipleAnswerQuestionOption"></param>
+        /// <returns></returns>
+        [Route("single-multiple-question")]
+        [HttpPost]
+        public IActionResult AddSingleMultipleAnswerQuestion(SingleMultipleAnswerQuestion singleMultipleAnswerQuestion, SingleMultipleAnswerQuestionOption singleMultipleAnswerQuestionOption)
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs]: how will you pass two objects?

File: Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs [hunk 5]
```diff
@@ -1,28 +1,30 @@
...
-        /// Gets all questions
+        /// Add single multiple answer question into model
         /// </summary>
-        /// <returns>Questions list</returns>
-        [HttpGet]
-        public IActionResult GetQuestions()
+        /// <param name="singleMultipleAnswerQuestion"></param>
+        /// <param name="singleMultipleAnswerQuestionOption"></param>
+        /// <returns></returns>
+        [Route("single-multiple-question")]
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Core/Controllers/QuestionsController.cs]: please remove dash from route

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionsRepository.cs (hunk 3 above)
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionsRepository.cs]: how will you add question table's id in its option table?

File: Trappist/src/Promact.Trappist.Repository/Questions/IQuestionRespository.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        /// </summary>
+        /// <returns>Question list</returns>
+        List<SingleMultipleAnswerQuestion> GetAllQuestions();
+
+        /// <summary>
+        /// Add single multiple answer question into model
+        /// </summary>
+        /// <param name="singleMultipleAnswerQuestion"></param>
+        /// <param name="singleMultipleAnswerQuestionOption"></param>
+        void AddSingleMultipleAnswerQuestion(SingleMultipleAnswerQuestion singleMultipleAnswerQuestion, SingleMultipleAnswerQuestionOption[] singleMultipleAnswerQuestionOption);
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/IQuestionRespository.cs]: please use list here

File: Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            var question = _dbContext.SingleMultipleAnswerQuestion.ToList();
+            return question;
+        }
+        /// <summary>
+        /// Add single multiple answer question into model
+        /// </summary>
+        /// <param name="singleMultipleAnswerQuestion"></param>
+        /// <param name="singleMultipleAnswerQuestionOption"></param>
+        public void AddSingleMultipleAnswerQuestion(SingleMultipleAnswerQuestion singleMultipleAnswerQuestion, SingleMultipleAnswerQuestionOption[] singleMultipleAnswerQuestionOption)
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs]: please use list here

**Pull request merged**